src/RouteRegistrar.php

```diff
@@ -162,13 +162,13 @@ protected function registerRoutes(ReflectionClass $class, ClassRouteAttributes $
                     ? $class->getName()
                     : [$class->getName(), $method->getName()];
 
-                $route = $this->router->addRoute($httpMethods, $attributeClass->uri, $action);
-
-                $route
-                    ->name($attributeClass->name);
-
-                if ($domain = $classRouteAttributes->domain()) {
-                    $route->domain($domain);
+                if ($domain = $classRouteAttributes->domainFromConfig() ?? $classRouteAttributes->domain()) {
+                    $route = $this->router->domain($domain)
+                        ->name($attributeClass->name)
+                        ->match($httpMethods, $attributeClass->uri, $action);
+                } else {
```

Comment:
Could this code be refactor to avoid this else?

Reply:
Yes, just refactored